Strange failure in DebugPrint test

### Description

We observed strange failure in `DebugPrint` memory hook.

https://ci.preferred.jp/cupy.win.cuda124/190640/392319

```
=========================== short test summary info ===========================
FAILED cupy_tests\cuda_tests\memory_hooks_tests\test_debug_print.py::TestDebugPrintHook::test_print - AssertionError: assert {'device_id':...em_size': 512} == {'device_id':...ze': 512, ...}
  
  Omitting 2 identical items, use -vv to show
  Differing items:
  {'hook': 'alloc'} != {'hook': 'free'}
  {'mem_ptr': 51640554496} != {'mem_ptr': 52357703680}
  Right contains 1 more item:
  {'pmem_id': '0x229af2537d0'}
  
  Full diff:
    {
        'device_id': 0,
  -     'hook': 'free',
  ?              ^^^^
  +     'hook': 'alloc',
  ?              ^^^^^
  -     'mem_ptr': 52357703680,
  +     'mem_ptr': 51640554496,
        'mem_size': 512,
  -     'pmem_id': '0x229af2537d0',
    }
= 1 failed, 106131 passed, 8346 skipped, 767 deselected, 290 xfailed, 2316 warnings in 8293.45s (2:18:13) =
```



### To Reproduce

```py
# Write the code here
```


### Installation

Wheel (`pip install cupy-***`)

### Environment

```
# Paste the output here
```


### Additional Information

_No response_

--- a/tests/cupy_tests/cuda_tests/memory_hooks_tests/test_debug_print.py
+++ b/tests/cupy_tests/cuda_tests/memory_hooks_tests/test_debug_print.py
@@ -1,3 +1,4 @@
+import gc
 import io
 import json
 import unittest
@@ -10,6 +11,7 @@ from cupy.cuda import memory_hooks
 class TestDebugPrintHook(unittest.TestCase):
 
     def setUp(self):
+        gc.collect()
         self.io = io.StringIO()
         self.hook = memory_hooks.DebugPrintHook(file=self.io)
         self.pool = memory.MemoryPool()

--- a/tests/cupy_tests/cuda_tests/memory_hooks_tests/test_line_profile.py
+++ b/tests/cupy_tests/cuda_tests/memory_hooks_tests/test_line_profile.py
@@ -1,3 +1,4 @@
+import gc
 import io
 import unittest
 import re
@@ -9,6 +10,7 @@ from cupy.cuda import memory_hooks
 class TestLineProfileHook(unittest.TestCase):
 
     def setUp(self):
+        gc.collect()
         self.pool = memory.MemoryPool()
 
     def test_print_report(self):

--- a/tests/cupy_tests/cuda_tests/test_memory_hook.py
+++ b/tests/cupy_tests/cuda_tests/test_memory_hook.py
@@ -1,3 +1,4 @@
+import gc
 import unittest
 
 import cupy.cuda
@@ -38,6 +39,7 @@ class SimpleMemoryHook(memory_hook.MemoryHook):
 class TestMemoryHook(unittest.TestCase):
 
     def setUp(self):
+        gc.collect()
         self.pool = memory.MemoryPool()
         self.unit = 512
 

